pkg/controller/dashboards/dashboard_controller.go

```diff
@@ -0,0 +1,160 @@
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	manifest = "bindata/dashboards/configmaps.yaml"
+)
+
+var (
+	//go:embed netstats.json
+	netstatsContent string
+	//go:embed ovn-health.json
+	ovnHealthContent string
+	dashboardRefs    []dashboardRef = []dashboardRef{
+		{
+			name:      "grafana-dashboard-network-stats",
+			json:      netstatsContent,
+			tplSuffix: "NetStats",
+		},
+		{
+			name:      "grafana-dashboard-ovn-health",
+			json:      ovnHealthContent,
+			tplSuffix: "OVNHealth",
+		},
+	}
+)
+
+type dashboardRef struct {
+	name      string
+	json      string
+	tplSuffix string
+}
+
+func Add(mgr manager.Manager, status *statusmanager.StatusManager, c cnoclient.Client) error {
+	return add(mgr, newReconciler(mgr, status, c))
+}
+
+func newReconciler(mgr manager.Manager, status *statusmanager.StatusManager, c cnoclient.Client) *ReconcileDashboard {
+	return &ReconcileDashboard{client: c, scheme: mgr.GetScheme(), status: status}
+}
+
+func add(mgr manager.Manager, r *ReconcileDashboard) error {
+	c, err := controller.New("dashboard-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// watch for changes in all configmaps in our namespace
+	cmInformer := v1coreinformers.NewConfigMapInformer(
+		r.client.Default().Kubernetes(),
+		names.DashboardNamespace,
+		0, // don't resync
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+
+	r.client.Default().AddCustomInformer(cmInformer) // Tell the ClusterClient about this informer
+
+	firstRun := true
+	return c.Watch(&source.Informer{Informer: cmInformer},
+		&handler.EnqueueRequestForObject{},
+		predicate.ResourceVersionChangedPredicate{},
+		predicate.NewPredicateFuncs(func(object crclient.Object) bool {
+			if firstRun {
+				firstRun = false
+				return true
+			}
+			for _, ref := range dashboardRefs {
+				if object.GetName() == ref.name {
+					return true
+				}
+			}
+			return false
+		}),
+	)
+}
+
+var _ reconcile.Reconciler = &ReconcileDashboard{}
+
+type ReconcileDashboard struct {
+	client cnoclient.Client
+	scheme *runtime.Scheme
+	status *statusmanager.StatusManager
+}
+
+func (r *ReconcileDashboard) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	klog.Infof("Reconcile dashboards")
+
+	// Fetch the Network.operator.openshift.io instance to get Network Type
+	operConfig := &operv1.Network{TypeMeta: metav1.TypeMeta{APIVersion: operv1.GroupVersion.String(), Kind: "Network"}}
+	err := r.client.Default().CRClient().Get(ctx, types.NamespacedName{Name: names.CLUSTER_CONFIG}, operConfig)
+	if err != nil {
+		err = fmt.Errorf("unable to retrieve Network.operator.openshift.io object: %w", err)
```

Comment:
nit: Can we start the error with a capital letter as outputting to klog usually starts with a capital letter? I assume its fine for the degraded message too. Same below.